CMakeLists.txt

```diff
@@ -1,83 +1,50 @@
-project(mallocMC LANGUAGES CUDA CXX)
+project(mallocMC LANGUAGES CXX)
 cmake_minimum_required(VERSION 3.8)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# helper for libs and packages
-set(CMAKE_CUDA_STANDARD 11)
-set(CMAKE_CUDA_STANDARD_REQUIRED ON)
-set(CMAKE_PREFIX_PATH "/usr/lib/x86_64-linux-gnu/"
-    "$ENV{CUDA_ROOT}" "$ENV{BOOST_ROOT}")
-
-
-################################################################################
-# CMake policies
-#
-# Search in <PackageName>_ROOT:
-#   https://cmake.org/cmake/help/v3.12/policy/CMP0074.html
-################################################################################
-
 if(POLICY CMP0074)
-  cmake_policy(SET CMP0074 NEW)
-endif()
-
-
-###############################################################################
-# CUDA
-###############################################################################
-if(NOT DEFINED COMPUTE_CAPABILITY)
-  set(COMPUTE_CAPABILITY "30")
-endif()
-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -arch=sm_${COMPUTE_CAPABILITY} -use_fast_math")
-
-OPTION(CUDA_OUTPUT_INTERMEDIATE_CODE "Output ptx code" OFF)
-if(CUDA_OUTPUT_INTERMEDIATE_CODE)
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -Xptxas -v --keep")
-endif()
-
-SET(CUDA_OPTIMIZATION_TYPE "unset" CACHE STRING "CUDA Optimization")
-set_property(CACHE CUDA_OPTIMIZATION_TYPE PROPERTY STRINGS "unset;-G0;-O0;-O1;-O2;-O3")
-if(NOT ${CUDA_OPTIMIZATION_TYPE} STREQUAL "unset")
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} ${CUDA_OPTIMIZATION_TYPE}")
+    cmake_policy(SET CMP0074 NEW)
 endif()
 
+# find alpaka
+set(ALPAKA_ROOT "${CMAKE_CURRENT_LIST_DIR}/alpaka/lib/cmake/alpaka" CACHE STRING "The location of alpakaConfig.cmake")
```

Comment:
Please have a look to https://github.com/alpaka-group/cupla/blob/b72833cbf0844664b26bbf553289356cfa03c18a/cuplaConfig.cmake#L112-L127
This is the way how we integrate alpaka in cupla. It gives you also easily a way to switch away from the internal shipped alpaka implementation.

Reply:
`add_subdirectory` is the way to use a internal version. An external will be used via `find_package`.

Reply:
Thank you! I added alpaka develop as a git subtree and took the cmake code from cupla.